### PR TITLE
Util for advanced toString building + log the detail of the PojoTypeManagers

### PR DESCRIPTION
**Pull request by yrodiere**
Example logs produced for `JavaBeanAnnotationMappingIT`:

```
10:17:08,593 (main) DEBUG PojoTypeManagerBuilder:78 - HSEARCH-POJO000017: Created POJO type manager: {
	indexedJavaClass=class org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity
	identifierMapping=org.hibernate.search.v6poc.entity.pojo.mapping.impl.PropertyIdentifierMapping@d2cc05a
	routingKeyProvider=org.hibernate.search.v6poc.entity.pojo.mapping.impl.RoutingKeyProvider$$Lambda$109/559670971@4f933fd1
	processor={
		class=PojoIndexingProcessorTypeNode
		objectAccessors=[]
		bridges=[
		]
		propertyNodes=[
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.time.LocalDate org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$ParentIndexedEntity.getLocalDate()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorValueBridgeNode
						bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultLocalDateValueBridge@548a9f61
						indexFieldAccessor=StubIncludedIndexFieldAccessor[myLocalDateField]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.util.OptionalInt org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity.getOptionalInt()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorValueBridgeNode
						bridge=org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.OptionalIntAsStringValueBridge@1753acfe
						indexFieldAccessor=StubIncludedIndexFieldAccessor[optionalIntAsString]
					}
					{
						class=PojoIndexingProcessorContainerElementNode
						extractor=org.hibernate.search.v6poc.entity.pojo.extractor.builtin.OptionalIntValueExtractor@7c16905e
						nestedNodes=[
							{
								class=PojoIndexingProcessorValueBridgeNode
								bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultIntegerValueBridge@2a2d45ba
								indexFieldAccessor=StubIncludedIndexFieldAccessor[optionalInt]
							}
						]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$ParentIndexedEntity.getEmbedded()]
				bridges=[
					org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.CustomPropertyBridge@2a5ca609
				]
				nestedNodes=[
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.lang.Integer[] org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity.getNumericArray()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorContainerElementNode
						extractor=org.hibernate.search.v6poc.entity.pojo.extractor.builtin.ArrayElementExtractor@20e2cbe0
						nestedNodes=[
							{
								class=PojoIndexingProcessorValueBridgeNode
								bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultIntegerValueBridge@68be2bc2
								indexFieldAccessor=StubIncludedIndexFieldAccessor[numericArray]
							}
						]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.lang.Iterable org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity.getEmbeddedIterable()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorContainerElementNode
						extractor=org.hibernate.search.v6poc.entity.pojo.extractor.builtin.IterableElementExtractor@28feb3fa
						nestedNodes=[
							{
								class=PojoIndexingProcessorTypeNode
								objectAccessors=[StubIncludedIndexObjectFieldAccessor[embeddedIterable]]
								bridges=[
								]
								propertyNodes=[
									{
										class=PojoIndexingProcessorPropertyNode
										handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getEmbedded()]
										bridges=[
										]
										nestedNodes=[
											{
												class=PojoIndexingProcessorTypeNode
												objectAccessors=[StubIncludedIndexObjectFieldAccessor[embeddedIterable.embedded]]
												bridges=[
												]
												propertyNodes=[
													{
														class=PojoIndexingProcessorPropertyNode
														handle=MemberPropertyHandle[public java.lang.String org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getText()]
														bridges=[
														]
														nestedNodes=[
															{
																class=PojoIndexingProcessorValueBridgeNode
																bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultStringValueBridge@675d3402
																indexFieldAccessor=StubIncludedIndexFieldAccessor[embeddedIterable.embedded.prefix_myTextField]
															}
														]
													}
												]
											}
										]
									}
								]
							}
						]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.util.Optional org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity.getOptionalText()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorContainerElementNode
						extractor=org.hibernate.search.v6poc.entity.pojo.extractor.builtin.OptionalValueExtractor@51565ec2
						nestedNodes=[
							{
								class=PojoIndexingProcessorValueBridgeNode
								bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultStringValueBridge@482f8f11
								indexFieldAccessor=StubIncludedIndexFieldAccessor[optionalText]
							}
						]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.util.Map org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity.getEmbeddedMap()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorContainerElementNode
						extractor=[org.hibernate.search.v6poc.entity.pojo.extractor.builtin.MapValueExtractor@1593948d, org.hibernate.search.v6poc.entity.pojo.extractor.builtin.CollectionElementExtractor@1b604f19]
						nestedNodes=[
							{
								class=PojoIndexingProcessorTypeNode
								objectAccessors=[StubIncludedIndexObjectFieldAccessor[embeddedMap]]
								bridges=[
								]
								propertyNodes=[
									{
										class=PojoIndexingProcessorPropertyNode
										handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getEmbedded()]
										bridges=[
										]
										nestedNodes=[
											{
												class=PojoIndexingProcessorTypeNode
												objectAccessors=[StubIncludedIndexObjectFieldAccessor[embeddedMap.embedded]]
												bridges=[
												]
												propertyNodes=[
													{
														class=PojoIndexingProcessorPropertyNode
														handle=MemberPropertyHandle[public java.time.LocalDate org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$ParentIndexedEntity.getLocalDate()]
														bridges=[
														]
														nestedNodes=[
															{
																class=PojoIndexingProcessorValueBridgeNode
																bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultLocalDateValueBridge@7823a2f9
																indexFieldAccessor=StubIncludedIndexFieldAccessor[embeddedMap.embedded.prefix_myLocalDateField]
															}
														]
													}
												]
											}
										]
									}
								]
							}
						]
					}
					{
						class=PojoIndexingProcessorContainerElementNode
						extractor=org.hibernate.search.v6poc.entity.pojo.extractor.builtin.MapKeyExtractor@4cc0edeb
						nestedNodes=[
							{
								class=PojoIndexingProcessorValueBridgeNode
								bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultStringValueBridge@457e2f02
								indexFieldAccessor=StubIncludedIndexFieldAccessor[embeddedMapKeys]
							}
						]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.util.List org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity.getEmbeddedList()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorContainerElementNode
						extractor=org.hibernate.search.v6poc.entity.pojo.extractor.builtin.CollectionElementExtractor@5c7fa833
						nestedNodes=[
							{
								class=PojoIndexingProcessorTypeNode
								objectAccessors=[StubIncludedIndexObjectFieldAccessor[embeddedList]]
								bridges=[
								]
								propertyNodes=[
									{
										class=PojoIndexingProcessorPropertyNode
										handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getEmbedded()]
										bridges=[
										]
										nestedNodes=[
											{
												class=PojoIndexingProcessorTypeNode
												objectAccessors=[StubIncludedIndexObjectFieldAccessor[embeddedList.otherPrefix_embedded]]
												bridges=[
													org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.CustomTypeBridge@39aeed2f
												]
												propertyNodes=[
												]
											}
										]
									}
								]
							}
						]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.lang.Integer org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity.getNumeric()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorValueBridgeNode
						bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultIntegerValueBridge@724af044
						indexFieldAccessor=StubIncludedIndexFieldAccessor[numeric]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.util.ArrayList org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$YetAnotherIndexedEntity.getEmbeddedArrayList()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorContainerElementNode
						extractor=org.hibernate.search.v6poc.entity.pojo.extractor.builtin.CollectionElementExtractor@4678c730
						nestedNodes=[
							{
								class=PojoIndexingProcessorTypeNode
								objectAccessors=[StubIncludedIndexObjectFieldAccessor[embeddedArrayList]]
								bridges=[
								]
								propertyNodes=[
									{
										class=PojoIndexingProcessorPropertyNode
										handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getEmbedded()]
										bridges=[
										]
										nestedNodes=[
											{
												class=PojoIndexingProcessorTypeNode
												objectAccessors=[StubIncludedIndexObjectFieldAccessor[embeddedArrayList.embedded]]
												bridges=[
												]
												propertyNodes=[
													{
														class=PojoIndexingProcessorPropertyNode
														handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$ParentIndexedEntity.getEmbedded()]
														bridges=[
															org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.CustomPropertyBridge@6767c1fc
														]
														nestedNodes=[
														]
													}
												]
											}
										]
									}
								]
							}
						]
					}
				]
			}
		]
	}
	indexManager=SimplifyingIndexManager[StubIndexManager[yetanotherindexedentity]]
}
10:17:08,597 (main) DEBUG PojoTypeManagerBuilder:78 - HSEARCH-POJO000017: Created POJO type manager: {
	indexedJavaClass=class org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$OtherIndexedEntity
	identifierMapping=org.hibernate.search.v6poc.entity.pojo.mapping.impl.PropertyIdentifierMapping@475530b9
	routingKeyProvider=org.hibernate.search.v6poc.entity.pojo.mapping.impl.RoutingKeyProvider$$Lambda$109/559670971@4f933fd1
	processor={
		class=PojoIndexingProcessorTypeNode
		objectAccessors=[]
		bridges=[
		]
		propertyNodes=[
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.lang.Integer org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$OtherIndexedEntity.getNumeric()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorValueBridgeNode
						bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultIntegerValueBridge@1d057a39
						indexFieldAccessor=StubIncludedIndexFieldAccessor[numeric]
					}
					{
						class=PojoIndexingProcessorValueBridgeNode
						bridge=org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.IntegerAsStringValueBridge@26be92ad
						indexFieldAccessor=StubIncludedIndexFieldAccessor[numericAsString]
					}
				]
			}
		]
	}
	indexManager=SimplifyingIndexManager[StubIndexManager[otherindexedentity]]
}
10:17:08,598 (main) DEBUG PojoTypeManagerBuilder:78 - HSEARCH-POJO000017: Created POJO type manager: {
	indexedJavaClass=class org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity
	identifierMapping=org.hibernate.search.v6poc.entity.pojo.mapping.impl.PropertyIdentifierMapping@224edc67
	routingKeyProvider=org.hibernate.search.v6poc.entity.pojo.mapping.impl.RoutingKeyProvider$$Lambda$109/559670971@4f933fd1
	processor={
		class=PojoIndexingProcessorTypeNode
		objectAccessors=[]
		bridges=[
			org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.CustomTypeBridge@14acaea5
		]
		propertyNodes=[
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.time.LocalDate org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$ParentIndexedEntity.getLocalDate()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorValueBridgeNode
						bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultLocalDateValueBridge@46d56d67
						indexFieldAccessor=StubIncludedIndexFieldAccessor[myLocalDateField]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$ParentIndexedEntity.getEmbedded()]
				bridges=[
					org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.CustomPropertyBridge@d8355a8
				]
				nestedNodes=[
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getEmbedded()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorTypeNode
						objectAccessors=[StubIncludedIndexObjectFieldAccessor[embedded]]
						bridges=[
							org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.CustomTypeBridge@59fa1d9b
						]
						propertyNodes=[
							{
								class=PojoIndexingProcessorPropertyNode
								handle=MemberPropertyHandle[public java.time.LocalDate org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$ParentIndexedEntity.getLocalDate()]
								bridges=[
								]
								nestedNodes=[
									{
										class=PojoIndexingProcessorValueBridgeNode
										bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultLocalDateValueBridge@28d25987
										indexFieldAccessor=StubIncludedIndexFieldAccessor[embedded.prefix_myLocalDateField]
									}
								]
							}
							{
								class=PojoIndexingProcessorPropertyNode
								handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$ParentIndexedEntity.getEmbedded()]
								bridges=[
									org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.CustomPropertyBridge@4501b7af
								]
								nestedNodes=[
								]
							}
							{
								class=PojoIndexingProcessorPropertyNode
								handle=MemberPropertyHandle[public org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getEmbedded()]
								bridges=[
								]
								nestedNodes=[
									{
										class=PojoIndexingProcessorTypeNode
										objectAccessors=[StubIncludedIndexObjectFieldAccessor[embedded.prefix_embedded]]
										bridges=[
											org.hibernate.search.v6poc.integrationtest.mapper.pojo.bridge.CustomTypeBridge@523884b2
										]
										propertyNodes=[
										]
									}
								]
							}
							{
								class=PojoIndexingProcessorPropertyNode
								handle=MemberPropertyHandle[public java.lang.String org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getText()]
								bridges=[
								]
								nestedNodes=[
									{
										class=PojoIndexingProcessorValueBridgeNode
										bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultStringValueBridge@5b275dab
										indexFieldAccessor=StubIncludedIndexFieldAccessor[embedded.prefix_myTextField]
									}
								]
							}
						]
					}
				]
			}
			{
				class=PojoIndexingProcessorPropertyNode
				handle=MemberPropertyHandle[public java.lang.String org.hibernate.search.v6poc.integrationtest.mapper.pojo.JavaBeanAnnotationMappingIT$IndexedEntity.getText()]
				bridges=[
				]
				nestedNodes=[
					{
						class=PojoIndexingProcessorValueBridgeNode
						bridge=org.hibernate.search.v6poc.entity.pojo.bridge.builtin.impl.DefaultStringValueBridge@61832929
						indexFieldAccessor=StubIncludedIndexFieldAccessor[myTextField]
					}
				]
			}
		]
	}
	indexManager=SimplifyingIndexManager[StubIndexManager[indexedentity]]
}
```

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/v6poc/logging/impl/ToStringTreeAppendableMultilineFormatter.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+package org.hibernate.search.v6poc.logging.impl;
+
+import org.hibernate.search.v6poc.util.spi.ToStringStyle;
+import org.hibernate.search.v6poc.util.spi.ToStringTreeAppendable;
+import org.hibernate.search.v6poc.util.spi.ToStringTreeBuilder;
+
+/**
+ * Used with JBoss Logging's {@link org.jboss.logging.annotations.FormatWith} to display
+ * {@link org.hibernate.search.v6poc.util.spi.ToStringTreeAppendable} objcects in log messages.
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/search/v6poc/logging/impl/ToStringTreeAppendableMultilineFormatter.java]: s/objcects/objects/

File: mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/impl/PojoTypeManager.java [hunk 2]
```diff
@@ -54,6 +61,15 @@ public void close() {
 		}
 	}
 
+	@Override
+	public void appendTo(ToStringTreeBuilder builder) {
+		builder.attribute( "indexedJavaClass", indexedJavaClass )
+				.attribute( "identifierMapping", identifierMapping )
+				.attribute( "routingKeyProvider", routingKeyProvider )
+				.attribute( "processor", processor )
+				.attribute( "indexManager", indexManager );
```
Comment (gsmet) [inline mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/impl/PojoTypeManager.java]: Let's put the index manager at the top. It's the most global object and the processor can be multiline so the index manager ends up mostly unnoticed.

Comment (yrodiere) [reply mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/impl/PojoTypeManager.java]: I was thinking about putting the details of the index managers (schema) at some point in the future, which would be very verbose. But for now you're right, it makes more sense to put it at the top, I'll do that.

File: engine/src/main/java/org/hibernate/search/v6poc/logging/impl/ToStringTreeAppendableMultilineFormatter.java (hunk 1 above)
Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/v6poc/logging/impl/ToStringTreeAppendableMultilineFormatter.java]: Fixed, thanks

**Comment by yrodiere** [comment]
Fixed the issues you mentioned and merged. Thanks!

**Pull request merged**